src/test/java/StationsTest.java

```diff
@@ -47,14 +47,30 @@ public void checkConnectingStations() throws IOException {
         Stations stations = givenStations();
 
         //when
-        stations.features.get(0).properties.getConnectingStations();
-        List<Integer> connectingStations = stations.features.get(0).properties.connectingStations;
+        stations.features.get(0).properties.findConnectingStations(stations);
+        List<Stations.Station> connectingStations = stations.features.get(0).properties.connectingStations;
 
         //then
         assertEquals("Astor Pl", stations.features.get(0).properties.name);
         assertNotNull(connectingStations);
-        assertTrue(connectingStations.contains(457));
-        assertTrue(connectingStations.contains(105));
+        assertTrue(connectingStations.contains(stations.findStation(457)));
+        assertTrue(connectingStations.contains(stations.findStation(105)));
+    }
+
+    @Test
+    public void checkShortestPath() throws IOException {
+        //given
+        Stations stations = givenStations();
+
+        //when
+        Path path1 = stations.findShortestPath(stations.findStation(32), stations.findStation(105));
+        Path path2 = stations.findShortestPath(stations.findStation(55), stations.findStation(186));
+        Path path3 = stations.findShortestPath(stations.findStation(12), stations.findStation(302));
+
+        //then
+        assertEquals(2, path1.length);
+        assertEquals(1, path2.length);
```

Comment:
I don't think you can have a path of one station since it should include the start and end station

Reply:
The length is one because it is the distance between the 2 stations, but there are 2 stations on the path.